crates/nu-command/src/date/utils.rs

```diff
@@ -1,43 +1,44 @@
+use chrono::{DateTime, FixedOffset, Local, LocalResult, TimeZone};
 use nu_protocol::{ShellError, Span, Value};
 
 pub fn unsupported_input_error(span: Span) -> Value {
     Value::Error {
-        error: ShellError::UnsupportedInput(
-            String::from(
-                "Only dates with timezones are supported. The following formats are allowed \n
-            * %Y-%m-%d %H:%M:%S %z -- 2020-04-12 22:10:57 +02:00 \n
-            * %Y-%m-%d %H:%M:%S%.6f %z -- 2020-04-12 22:10:57.213231 +02:00 \n
-            * rfc3339 -- 2020-04-12T22:10:57+02:00 \n
-            * rfc2822 -- Tue, 1 Jul 2003 10:52:37 +0200",
-            ),
-            span,
-        ),
+        error: ShellError::UnsupportedInput(String::from("Unable to parse into datetime."), span),
     }
 }
 
-pub fn parse_date_from_string(input: String, span: Span) -> Result<DateTime<FixedOffset>, Value> {
-    let datetime = DateTime::parse_from_str(&input, "%Y-%m-%d %H:%M:%S %z"); // "2020-04-12 22:10:57 +02:00";
-    match datetime {
-        Ok(x) => Ok(x),
-        Err(_) => {
-            let datetime = DateTime::parse_from_str(&input, "%Y-%m-%d %H:%M:%S%.6f %z"); // "2020-04-12 22:10:57.213231 +02:00";
-            match datetime {
-                Ok(x) => Ok(x),
-                Err(_) => {
-                    let datetime = DateTime::parse_from_rfc3339(&input); // "2020-04-12T22:10:57+02:00";
-                    match datetime {
-                        Ok(x) => Ok(x),
-                        Err(_) => {
-                            let datetime = DateTime::parse_from_rfc2822(&input); // "Tue, 1 Jul 2003 10:52:37 +0200";
-                            match datetime {
-                                Ok(x) => Ok(x),
-                                Err(_) => Err(unsupported_input_error(span)),
-                            }
-                        }
-                    }
+pub(crate) fn parse_date_from_string(
+    input: &str,
+    span: Span,
+) -> Result<DateTime<FixedOffset>, Value> {
+    match dtparse::parse(input) {
+        Ok((native_dt, fixed_offset)) => {
+            let offset = match fixed_offset {
+                Some(fo) => fo,
+                None => *(Local::now().offset()),
+            };
+            match offset.from_local_datetime(&native_dt) {
+                LocalResult::Single(d) => Ok(d),
+                LocalResult::Ambiguous(d, _) => Ok(d),
+                LocalResult::None => {
+                    Err(Value::Error {
+                        error: ShellError::CantConvert(
+                            "could not convert to a timezone-aware datetime"
+                                .to_string(),
+                            "local time representation is invalid".to_string(),
+                            span,
+                        )
+                    })
                 }
             }
         }
+        Err(_) => {
+            Err(Value::Error {
+                error: ShellError::UnsupportedInput(
```

Comment:
We may want to add a new `ShellError` case for this that talks about date conversions, that has a notes field in the error where we can talk more about the kinds of date strings that are allowed (which was in some of the code you took out). This lets us have a nicer error, less duplication, but also lets us keep the helpful info for when we do have an error.